docs/guides/using-rewrites.md

```diff
@@ -0,0 +1,188 @@
+# Rewriting paths
+This guide demonstrates host and path rewrites using Ingress and Service configuration.
+
+## Installation
+
+Please follow the [deployment](../deployment) documentation to install
+Kong Ingress Controller on your Kubernetes cluster.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](../deployment) to configure this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Date: Fri, 21 Jun 2019 17:01:07 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Create a test Deployment
+
+To test our requests, we create an echo server Deployment, which responds to
+HTTP requests with a summary of the request contents:
+
+```bash
+$ kubectl create namespace echo
+namespace/echo created
+$ kubectl apply -n echo -f https://bit.ly/echo-service
+service/echo created
+deployment.apps/echo created
+```
+
+## Create a Kubernetes service
```

Comment:
`https://bit.ly/echo-service` creates a Kubernetes Service already.
Can we skip this step?

Reply:
Split on this--we later modify the service in question, so for the purposes of a detailed guide I think it's useful to walk users through creating it before doing so rather than conjuring it out of the ether. Leaving it in for now on the basis that it's at least not harmful, but changing to an apply to sneakily overwrite the original and at maintain naming consistency with other guides.

For things that the guide doesn't interact with directly, but does need to function (e.g. the Deployment in this case) it definitely makes sense to let the bootstrap file take care of it.

Reply:
Explaining and walking users through something is good.
This doesn't feel consistent with other guides but that's mostly my OCD kicking in.

Although, I'd like the bitly to then deploy only a Deployment. Deploying more resources than necessary confuses users. This can be improved upon later or be part of this PR. Either way works for me. Let me know.

Reply:
IMO this case is a good one to ack the OCD alarm but dismiss it--the `apply` in instructions versus the previous `create` means they only wind up with one service. Having multiple (with one unused) would be more confusing, as it's _creating_ things that aren't necessary.

I'm more okay with updating an existing one with an apply, since that's a normal Kubernetes workflow anyway, which is _doing_ something that's not strictly necessary, but not too out of the ballpark. The end state isn't abnormal, so I'm not as concerned. Deferring to a later PR if we still want to make changes after.